Vizzuality/gfw
# Feature/map search

## Overview

This pr adds new design update for the search panel. 


1. there are a few style changes for previous content, but basically kept the colors the same as defined colors where not in our config. 
2. Modified styles to make sure only search results are overflowed, otherwise the entire menu will move when results overflow the container
3. As previous statement, i decided to make sure the search box is always visible on desktop when scrolling, so you dont loose context of what you searched for. 
4. Mobile layout is handled mostly by flex logic, as we will only show eather menu/result container and let flex grow deal with the layout. For this change the logic became a bit more complex in the search component; so did some splitting of logic there to make it easier to work with in the future. 
5. Created a `vertical-menu` component that we can re-use when we need it, as of now its quite simple and we can extend if we need more logic there. 

## Demo

<img width="921" alt="Screenshot 2020-09-08 at 14 42 47" src="https://user-images.githubusercontent.com/971129/92477924-b08a2300-f1e1-11ea-95a1-57b7a1311630.png">


## Notes

We still use "one" component for datasets/locations, as the behaviour is still similar. and when switching between the location/dataset menu it behaves like a "filter" and you dont loose the previous context. 

...

## Testing

- Make sure search works as it was before, as nothing should have been changed there, but all old logic should still work. 
- Make sure navigation works as expected.
- Make sure styles are correct (as i didnt use the colors in (invision). 
- Tested in webkit,ff edge not sure if we need to check ie 11 ? 
- Make sure mobile/desktop layouts behave as expected both in reload/resize browser window.
